### PR TITLE
[JCLOUDS-846] Completing missing mock tests

### PR DESCRIPTION
**Pull request by ilgrosso**

### CODE REVIEW

File: azurecompute/src/main/java/org/jclouds/azurecompute/domain/StorageService.java [hunk 1]
```diff
@@ -16,49 +16,224 @@
...
 
+   /**
+    * Specifies the properties of the storage account.
+    */
    public abstract StorageServiceProperties storageServiceProperties();
 
+   /**
+    * Specifies the name and value of an extended property that was added to the storage account.
+    */
+   public abstract Map<String, String> extendedProperties();
```
Comment (nacx) [inline azurecompute/src/main/java/org/jclouds/azurecompute/domain/StorageService.java]: This seems to be nullable?

File: azurecompute/src/main/java/org/jclouds/azurecompute/domain/StorageService.java [hunk 2]
```diff
@@ -16,49 +16,224 @@
...
+
+      /**
+       * The status of the storage account.
+       */
+      public abstract Status status();
+
+      /**
+       * Specifies the endpoints of the storage account.
+       */
+      public abstract List<URL> endpoints();
```
Comment (nacx) [inline azurecompute/src/main/java/org/jclouds/azurecompute/domain/StorageService.java]: This seems to be nullable

File: azurecompute/src/test/java/org/jclouds/azurecompute/features/SubscriptionApiMockTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      try {
+         final SubscriptionApi api = api(server.getUrl("/")).getSubscriptionApi();
+
+         assertEquals(api.listRoleSizes(), ListRoleSizesHandlerTest.expected());
+
+         assertSent(server, "GET", "/rolesizes");
+      } finally {
+         server.shutdown();
+      }
+   }
```
Comment (nacx) [inline azurecompute/src/test/java/org/jclouds/azurecompute/features/SubscriptionApiMockTest.java]: Add also a test that exercises the fallback.

**Comment by nacx** [comment]
Thanks!

**Comment by ilgrosso** [comment]
@nacx Comments addressed and code rebased.

**Comment by nacx** [comment]
Squashed and pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/aac95744) and [1.9.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/cb5df375). Thanks!

**Pull request closed without merging**